src/functionalTest/java/uk/gov/hmcts/reform/locationrefdata/cameltest/LrdBatchApplicationTest.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import uk.gov.hmcts.reform.data.ingestion.configuration.AzureBlobConfig;
+import uk.gov.hmcts.reform.data.ingestion.configuration.StorageCredentials;
+import uk.gov.hmcts.reform.locationrefdata.camel.binder.ServiceToCcdCaseType;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.LrdBatchIntegrationSupport;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.RestartingSpringJUnit4ClassRunner;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.SpringRestarter;
+import uk.gov.hmcts.reform.locationrefdata.config.LrdCamelConfig;
+import uk.gov.hmcts.reform.locationrefdata.configuration.BatchConfig;
+
+import java.io.FileInputStream;
+
+import static org.springframework.util.ResourceUtils.getFile;
+
+@TestPropertySource(properties = {"spring.config.location=classpath:application-integration.yml,"
+    + "classpath:application-leaf-integration.yml"})
+@RunWith(RestartingSpringJUnit4ClassRunner.class)
+@MockEndpoints("log:*")
+@ContextConfiguration(classes = {LrdCamelConfig.class, CamelTestContextBootstrapper.class,
+    JobLauncherTestUtils.class, BatchConfig.class, AzureBlobConfig.class, StorageCredentials.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+@SpringBootTest
+@EnableAutoConfiguration(exclude = JpaRepositoriesAutoConfiguration.class)
+@EnableTransactionManagement
+@SqlConfig(dataSource = "dataSource", transactionManager = "txManager",
+    transactionMode = SqlConfig.TransactionMode.ISOLATED)
+@SuppressWarnings("unchecked")
+public class LrdBatchApplicationTest extends LrdBatchIntegrationSupport {
+
+    @Value("${start-route}")
+    private String startRoute;
+
+    @Value("${archival-route}")
+    String archivalRoute;
+
+    @Before
+    public void init() {
+        jdbcTemplate.execute(truncateAudit);
+        SpringRestarter.getInstance().restart();
+    }
+
+    @Test
+    @Sql(scripts = {"/testData/truncate-lrd.sql"})
+    public void testTaskletSuccess() throws Exception {
+        testInsertion();
+    }
+
+    @Test
+    @Sql(scripts = {"/testData/truncate-lrd.sql"})
+    public void testTaskletSuccessWithUpdateAndDelete() throws Exception {
+
+        testInsertion();
+        jdbcTemplate.execute("delete from DATALOAD_SCHEDULAR_AUDIT");
+        integrationTestSupport.uploadFile(
+            "service-test.csv",
+            new FileInputStream(getFile(
+                "classpath:sourceFiles/service-test-update.csv"))
+        );
```

Comment:
we should also include a scenario where we only have to delete and no insert

Reply:
can you put use case may be in jira? I am not clear what should excepted here.

Reply:
It is already included in the acceptance criteria under #c of 2 and  #3

Reply:
ok..This is just file name will rename this

Reply:
service-test-empty-case-or-name created new file